lgc/builder/ArithBuilder.cpp

```diff
@@ -1310,13 +1310,26 @@ Value *BuilderImpl::CreateFindSMsb(Value *value, const Twine &instName) {
   assert(value->getType()->getScalarType()->isIntegerTy(32));
 
   Constant *negOne = ConstantInt::get(value->getType(), -1);
-  Value *leadingZeroCount =
-      scalarize(value, [this](Value *value) { return CreateUnaryIntrinsic(Intrinsic::amdgcn_sffbh, value); });
+  Value *leadingZeroCount = CreateCountLeadingSignBits(value);
+  Value *isNegOne = CreateICmpEQ(leadingZeroCount, negOne);
```

Comment:
The name is misleading: should be leadingSign(Bits)Count. Rest LGTM

Reply:
updated. Thanks.